addons/mail/models/mail_render_mixin.py

```diff
@@ -318,7 +292,8 @@ def _render_template_qweb(self, template_src, model, res_ids,
         for record in self.env[model].browse(res_ids):
             variables['object'] = record
             try:
-                render_result = self.env['ir.qweb']._render(html.fragment_fromstring(template_src), variables)
+                render_result = self.env['ir.qweb']._render(lxml.html.fragment_fromstring(template_src, create_parent=True), variables)[5:-6].decode()
```

Comment:
What is this "[5:-6]" ? Are you trying to remove some extra tag ? In which case this should be done in a clean way, this is quite hackish. Indeed if I render QWeb I don't expect it to magically gain or loose tags, seems quite strange.

Reply:
The reason is that `lxml.html.fragment_fromstring` cannot have leading tags (eg. `<p>one</p><p>two</p>`).
`create_parent=True` wrap the content in a div systematically. Therefore I systematically remove it with [5: -6].

I could use `create_parent="t"` or `"<t>" + template_src + "</t>"` so qweb will remove the wrapped tag automatically. I will try the <t> solution and provide the result once I have it.

Reply:
That's indeed what we had in mind at first thoughts, trying to use the magic <t> :)

Reply:
It worked perfectly without issues.